src/gadgets/nonnative/bn/big_uint_mul_mod_chip/mod.rs

```diff
@@ -27,6 +27,10 @@ pub enum Error {
         "During the calculation of carry bits the number is converted to f64 and an error occurred"
     )]
     CarryBitsCalculate,
+    #[error("Error while calculate limbs per group count")]
+    ZeroLimbsPerGroup,
+    #[error("Carry bits equal capacity of `Curve::Base`")]
+    CarryBitsEqualCapacity,
 }
 
 pub const MAIN_GATE_T: usize = 4;
```

Comment:
No, that function is extremely general, it can be any size, but the chips that use the main gate config are already responsible for validating the original size